lib/Parser/Scan.cpp

```diff
@@ -1588,6 +1641,98 @@ tokens Scanner<EncodingPolicy>::ScanStringConstant(OLECHAR delim, EncodedCharPtr
 
 /*****************************************************************************
 *
+*  Scan annotations used for type hints and function ids.
+*  Pointer is expected to be inside a multiline comment
+*  Return tkNone if there are no annotations inside this comment
+*/
+template<typename EncodingPolicy>
+tokens Scanner<EncodingPolicy>::ScanAnnotations(EncodedCharPtr *pp)
+{
+	EncodedCharPtr p = *pp;
+	EncodedCharPtr last = m_pchLast;
+	OLECHAR ch;
+	//FCASTE: IMPORTANT
+	for (;;)
+	{
+		switch ((ch = this->ReadFirst(p, last)))
+		{
+		case '*':
+			if (*p == '/')
+			{
+				*pp = p + 1;
+				if (m_fSyntaxColor)
+				{
+					m_scanState = ScanStateNormal;
+					return tkComment;
+				}
+				return tkNone;
+			}
+			break;
+		case '@':
+			switch (p[0])
+			{
+			case 't':
+				if (p[1] == 'y' && p[2] == 'p' && p[3] == 'e' && !IsIdContinueNext(p + 4, last))
+				{
+                    //+5 so we skip the space (change that)
+					p += 5;//FCASTE: don't like this code, revisit 
+                    switch ((ch = this->ReadFirst(p, last))) 
+                    {
+                    case '{':
+                        *pp = p;
+                        return tkTypeAnnBegin;
+                    }
+				}
+				Error(ERRsyntax);
```

Comment:
Fair enough. In that case, we should add a runtime flag and put the type annotations behind it. Otherwise we will break any code that has @ inside of a block comment (looks like one of our UTs is hitting this issue).